src/bogart/AS_BAT_MarkRepeatReads.C

```diff
@@ -389,136 +364,56 @@ findThickestPrevRead(Unitig *tig, uint32 fi, int32 rMin, int32 rMax) {
     }
   }
 
-  //  Decide if the repeat region is to the left or right of this read.
-  //  Then ignore this thickest read if it doesn't fall in the repeat correctly.
-  //
-  //  A prev that ends inside the repeat, and so could be confused:
-  //
-  //              ----[rrrrrrrrrrrrrrrrrrrrrrrrrrrrr]------------------
-  //                                   ----------------------rdA
-  //                    rdB------------------
-  //
-  //  A prev that ends outside the repeat, can't be confused.
-  //
-  //              ----[rrrrrrrrrrrrrrrrrrrrrrrrrrrrr]------------------
-  //                                   ----------------------rdA
-  //                    rdB----------------------------
-  //
+  //  If this thickest overlap ends outside the repeat region, the overlap
+  //  cannot be confused.
 
-  assert(rdAlo <= rMax);
-  assert(rMin <= rdAlo);
+  ufNode *rdB   = nullptr;
+  int32   rdBlo = 0;
+  int32   rdBhi = 0;
 
-  int32    rdBlo = tig->ufpath[bestIdx].position.min();
-  int32    rdBhi = tig->ufpath[bestIdx].position.max();
+  if (bestIdx != UINT32_MAX) {
+    assert(rdAlo <= rMax);
+    assert(rMin <= rdAlo);
 
-  //  If the high coordinate isn't inside the repeat, we don't care;
-  //  rdB will be anchored to unique sequence.
-  //
-  //  For finding the previous read:
-  //
-  //        --------[repeat]]]--------
-  //                   -------------         READ WE'RE SEARCHING FROM
-  //                 ---------u              DON'T CARE, anchored in unique (u)
-  //          -----                          DON'T CARE, shouldn't happen
-  //
-  //  For the next read:
-  //
-  //        ------[[[repeat]----------
-  //       -------------                     READ WE'RE SEARCHING FROM
-  //             u---------                  DON'T CARE, anchored in unique (u)
-  //                          ------         DON'T CARE, shouldn't happen
-  //
-  //  The repeat is extended a bit so that the read will have some useful bit
-  //  of unique anchoring.
+    rdB   = &tig->ufpath[bestIdx];
+    rdBlo =  tig->ufpath[bestIdx].position.min();
+    rdBhi =  tig->ufpath[bestIdx].position.max();
 
-  if (isInside(rMin, rdBhi, rMax + 50) == false) {
-    bestId  = 0;
-    bestIdx = UINT32_MAX;
-    bestLen = 0;
+    if (isInside(rMin, rdBhi, rMax + 50) == false)
```

Comment:
You are not updating rdBlo/hi here, is it OK?

Reply:
rdBlo/rdBhi are only used if rdB isn't null.